spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetReaders.java

```diff
@@ -497,6 +514,48 @@ protected MapData buildMap(ReusableMapData map) {
     }
   }
 
+  /** Variant reader to convert from Variant to Spark VariantVal */
+  private static class VariantReader implements ParquetValueReader<VariantVal> {
```

Comment:
I think you could extend `DelegatingValueReader` to avoid needing to implement the full `ParquetValueReader` contract. You'd just need to pass the delegate in the constructor and then you could call `readFromDelegate` to produce the `Variant`.

Reply:
This is fine because `ParquetValueReader` already exposes `TripleIterator`. But at least we avoid using it.